src/components/Search/beta/SearchDropdown.tsx

```diff
@@ -20,12 +20,46 @@ import {
   DropDownMenuRaw,
   DropDownItemRaw,
 } from '../../../styles';
+import type { Repository } from '../../../types/state';
 
-function SearchDropdown(props: SearchDropdownProps) {
-  const { repository, view, selectLabel, tooltipText, mode } = props;
+function SearchDropdown({
+  repository,
+  view,
+  selectLabel,
+  tooltipText,
+  mode,
+  base,
+}: SearchDropdownProps) {
   const size = view == 'compare-results' ? 'small' : undefined;
   const { handleChangeDropdown } = useHandleChangeDropdown();
 
+  //searchType is to distinguish between base and new search dropdowns for handleChangeDropdown hook
+  const [repoSelect, setRepoSelect] = useState({
+    baseRepository: 'try',
+    newRepository: 'try',
+    searchType: base,
+  });
+
+  useEffect(() => {
+    void handleChangeDropdown(repoSelect);
+  }, [repoSelect]);
+
+  const handleRepoSelect = (e: React.MouseEvent<HTMLLIElement>) => {
+    const { classList, id } = e.currentTarget;
+    let name = '';
+    classList.forEach((item) => {
+      if (item.includes('Repository')) {
+        name = item;
+        return;
+      }
+    });
```

Comment:
This looks very brittle. Why don't you use `base` directly, to decide which repository to use?

Also it's not clear to me why we use `onClick` on the menu items instead of the property `onChange` on `<Select>`? Then the selected value can be retrieved in `event.target.value` (see https://mui.com/material-ui/api/select/).